src/main/java/bob/Bob.java

```diff
@@ -0,0 +1,355 @@
+        } catch (IndexOutOfBoundsException | NumberFormatException e) {
+            throw new BobException("The task number provided is invalid.");
+        }
+    }
+
+    /**
+     * Handles the "TODO" command to add a ToDo task.
+     *
+     * @param taskDetails The description of the ToDo task.
+     * @throws BobException If there are missing details.
+     */
+    static void commandTodo(String taskDetails) throws BobException {
+        String format = "Add ToDo task in the following format:\ntodo <description>";
+        if (taskDetails.isEmpty()) {
+            throw new BobException("Missing details!\n" + format);
+        }
+        // Create a new task
+        ToDo task = new ToDo(taskDetails);
+
+        // Add task and update tasks
+        tasks.addTask(task);
+        storage.saveTasks(tasks);
+        ui.showMessage("Adding ToDo task:\n " + task
+                + "\nTotal number of tasks in your list: " + tasks.getNumTasks());
+    }
+
+    /**
+     * Handles the "DEADLINE" command to add a Deadline task.
+     *
+     * @param taskDetails The description and due date of the Deadline task.
+     * @throws BobException If there are missing details, wrong format or the due date is invalid.
+     */
+    static void commandDeadline(String taskDetails) throws BobException {
+        String format = "Add Deadline task in the following format:\n"
+                + "deadline <description> /by <due date>";
+        if (taskDetails.isEmpty()) {
+            throw new BobException("Missing details!\n" + format);
+        }
+
+        try {
+            // Process task details
+            String[] params = taskDetails.split(" /");
+            String description = params[0];
+            String byStr = params[1].split(" ", 2)[1];
+
+            LocalDateTime by = Parser.parseDateTime(byStr);
+            if (by.isBefore(LocalDateTime.now())) {
+                throw new BobException("Oops! The date you provided is in the past. "
+                        + "Kindly provide a future date.");
+            }
+
+            // Create a new task
+            Deadline task = new Deadline(description, by);
+
+            // Add task and update tasks
+            tasks.addTask(task);
+            storage.saveTasks(tasks);
+            ui.showMessage("Adding Deadline task:\n " + task
+                    + "\nTotal number of tasks in your list: " + tasks.getNumTasks());
+        } catch (IndexOutOfBoundsException e) {
+            throw new BobException("You may have missing details or wrong format!\n" + format);
+        }
+    }
+
+    /**
+     * Handles the "EVENT" command to add an Event task.
+     *
+     * @param taskDetails The description, start date, and end date of the Event task.
+     * @throws BobException If there are missing details, wrong format, or invalid dates.
+     */
+    static void commandEvent(String taskDetails) throws BobException {
```

Comment:
maybe you can rename commandEvent to commandAddEvent to make it clearer as a method name?